src/table/utils/handle-scroll.js

```diff
@@ -1,11 +1,21 @@
-export const handleScroll = (evt, tableContainerRef) => {
+export const handleHorizontalScroll = (evt, rtl, memoedContainer) => {
   evt.stopPropagation();
-  let { scrollLeft } = tableContainerRef.current;
-  const newScrollLeft = scrollLeft + evt.deltaX;
-  const max = tableContainerRef.current.scrollWidth - tableContainerRef.current.offsetWidth;
-  if (max > 0 && (newScrollLeft < 0 || newScrollLeft > max)) {
-    evt.preventDefault();
-    scrollLeft = Math.max(0, Math.min(max, newScrollLeft));
+  if (rtl) {
+    const scrollLeft = memoedContainer.scrollLeft + evt.deltaX;
+    let scrollRight = memoedContainer.scrollWidth - (scrollLeft + memoedContainer.clientWidth);
+    const max = memoedContainer.scrollWidth + memoedContainer.offsetWidth;
+    if (max > 0 && (scrollRight < 0 || scrollRight > max)) {
+      evt.preventDefault();
+      scrollRight = Math.max(0, Math.min(max, scrollRight));
```

Comment:
How is this applied to anything? `scrollRight` isn't a property on `memoedContainer` or?

Reply:
Element only has [scrollLeft](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollLefts) unfortunately. 
It is only applied in RTL. When the length of scrollRight is larger than the available scroll length (which is the `max` variable), preventDefault is run to prevent going to the previous page.

Reply:
yeah but what happens when you actually set `scrollRight` in here, it is a local variable, then nothing is returned

Reply:
yes, good point. Something wrong here. It is weird that it worked before. let me fix it.